### PR TITLE
POC - use AbortController for managing event listeners

### PR DESCRIPTION
**Pull request by jamieomaguire**
Please click the `Preview` tab and select a different PR template if your change isn't for an                                            existing web component.

[PIE Docs Template](?expand=1&template=docs_template.md)
[New Web Component Template](?expand=1&template=new_component_template.md)

═══════════════════════════════════════════════════════════

## Author Checklist (complete before requesting a review)
- [ ] I have performed a self-review of my code
- [ ] I have added thorough tests where applicable (unit / component / visual)
- [ ] I have reviewed the `PIE Storybook`/`PIE Docs` PR preview
- [ ] I have reviewed visual test updates properly before approving
- [ ] If changes will affect consumers of the package, I have created a changeset entry.
- [ ] If a changeset file has been created, I have used the `/snapit` functionality to test my changes in a consuming application

### Reviewer 1
- [ ] I have reviewed the `PIE Storybook`/`PIE Docs` PR preview
- [ ] If there are visual test updates, I have reviewed them

### Reviewer 2
- [ ] I have reviewed the `PIE Storybook`/`PIE Docs` PR preview
- [ ] If there are visual test updates, I have reviewed them

### CODE REVIEW

File: packages/components/pie-modal/src/index.ts [hunk 1]
```diff
@@ -108,21 +108,36 @@ export class PieModal extends RtlMixin(LitElement) implements ModalProps {
...
+            signal: this._eventsListenerController.signal,
+        });
     }
 
     disconnectedCallback () : void {
-        document.removeEventListener(ON_MODAL_OPEN_EVENT, (event) => this._handleModalOpened(<CustomEvent>event));
-        document.removeEventListener(ON_MODAL_CLOSE_EVENT, (event) => this._handleModalClosed(<CustomEvent>event));
-        document.removeEventListener(ON_MODAL_BACK_EVENT, (event) => this._handleModalClosed(<CustomEvent>event));
+        this._eventsListenerController?.abort(); // Cleans up all added event listeners
+        this._eventsListenerController = null;
```
Comment (xander-marjoram) [inline packages/components/pie-modal/src/index.ts:139]: I reckon the GC would sort this out so might not be necessary to clean up beyond calling `abort`

File: packages/components/pie-modal/src/index.ts [hunk 2]
```diff
@@ -108,21 +108,36 @@ export class PieModal extends RtlMixin(LitElement) implements ModalProps {
...
         super.connectedCallback();
-        this.addEventListener('click', (event) => this._handleDialogLightDismiss(event));
-        document.addEventListener(ON_MODAL_OPEN_EVENT, (event) => this._handleModalOpened(<CustomEvent>event));
-        document.addEventListener(ON_MODAL_CLOSE_EVENT, (event) => this._handleModalClosed(<CustomEvent>event));
-        document.addEventListener(ON_MODAL_BACK_EVENT, (event) => this._handleModalClosed(<CustomEvent>event));
+        this._eventsListenerController = new AbortController();
+
+        this.addEventListener('click', (event) => this._handleDialogLightDismiss(event), {
+            signal: this._eventsListenerController.signal,
+        });
```
Comment (xander-marjoram) [inline packages/components/pie-modal/src/index.ts:122]: Just a formatting question for if we do decide to implement this, could we destructure `signal` first?

```suggestion
        this._eventsListenerController = new AbortController();
        const { signal } = this._eventsListenerController;

        this.addEventListener('click', (event) => this._handleDialogLightDismiss(event), { signal });
```

File: packages/components/pie-modal/src/index.ts [hunk 3]
```diff
@@ -132,9 +147,15 @@ export class PieModal extends RtlMixin(LitElement) implements ModalProps {
         if (this._dialog) {
             const dialogPolyfill = await import('dialog-polyfill').then((module) => module.default);
             dialogPolyfill.registerDialog(this._dialog);
-            this._dialog.addEventListener('cancel', (event) => this._handleDialogCancelEvent(event));
```
Comment (xander-marjoram) [inline packages/components/pie-modal/src/index.ts:135]: I guess we were never removing these event listeners before?

File: packages/components/pie-modal/src/index.ts (hunk 1 above)
Comment (jamieomaguire) [reply packages/components/pie-modal/src/index.ts:139]: yup that's probably a good shout!

File: packages/components/pie-modal/src/index.ts (hunk 2 above)
Comment (jamieomaguire) [reply packages/components/pie-modal/src/index.ts:122]: can do!

**Pull request closed without merging**